carp.common/src/commonTest/kotlin/dk/cachet/carp/common/MACAddressTest.kt

```diff
@@ -0,0 +1,39 @@
+package dk.cachet.carp.common
+
+import dk.cachet.carp.common.serialization.createDefaultJSON
+import kotlin.test.*
+
+
+/**
+ * Tests for [MACAddress].
+ */
+class MACAddressTest
+{
+    @Test
+    fun can_serialize_and_deserialize_MACAddress_using_JSON()
+    {
+        val mac = MACAddress( "00-11-22-33-44-55" )
+
+        val json = createDefaultJSON()
+        val serialized = json.encodeToString( MACAddress.serializer(), mac )
+        val parsed = json.decodeFromString( MACAddress.serializer(), serialized )
+
+        assertEquals( mac, parsed )
+    }
+
+    @Test
+    fun cant_initialize_incorrect_MACAddress()
```

Comment:
Tests should be updated if you agree to the more relaxed format

Reply:
I agree to a more relaxed format. But introducing that would be a backwards compatible change. It is simply something I did not want to spend time on now. If you think it is essential, I can work on it as part of this release.

Reply:
Either way you need this test case:
`assertFailsWith<IllegalArgumentException> { MACAddress( "hey you and the rock-steady AB-CD-EF-12-34-45 crew" ) }`

Reply:
...of course with no valid delimiter on either side of the mac. So get rid of the hyphen :)

Reply:
assertFailsWith<IllegalArgumentException> { MACAddress( "... AB-CD-EF-12-34-56 ..." ) } passes. `Regex.matches` always verifies the _entire_ input: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/-regex/matches.html

Reply:
Ok interesting semantics there.

Reply:
https://www.baeldung.com/kotlin-regular-expressions This is the difference between `containsMatchIn` and `matches`. I am using `matches`, so it works as expected. No bug. :)